src/fw-update/fw-update-device-interface.h

```diff
@@ -9,18 +9,15 @@
 
 namespace librealsense
 {
-    class updatable
+    class fw_checkable
     {
     public:
-        virtual void enter_update_state() const = 0;
-        virtual std::vector<uint8_t> backup_flash(update_progress_callback_ptr callback) = 0;
-        virtual void update_flash(const std::vector<uint8_t>& image, update_progress_callback_ptr callback, int update_mode) = 0;
         virtual bool check_fw_compatibility(const std::vector<uint8_t>& image) const = 0;
-        std::string extract_firmware_version_string(const void* fw_image, size_t fw_image_size) const
+        static std::string extract_firmware_version_string(const void* fw_image, size_t fw_image_size)
```

Comment:
I think you can pass it in as a `vector<uint8_t> vector` instead of C-types

Reply:
done